### PR TITLE
FT2023-343: Review block tasks

### PR DESCRIPTION
**Pull request by Eshakundu01**
The module name is **user_role**
- Create a custom block which will display “Welcome [User Role]” on the top region available
- The block will be displayed on the page /custom-welcome-page
- The user once successfully logs in will be redirected to this above-mentioned page.

### CODE REVIEW

File: web/modules/custom/user_role/src/Plugin/Block/UserRoleBlock.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    $instance = new static($configuration, $plugin_id, $plugin_definition);
+    $instance->user = $container->get('current_user');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $role = join(",", $this->user->getRoles(TRUE));
```
Comment (vivekpan27) [inline web/modules/custom/user_role/src/Plugin/Block/UserRoleBlock.php:44]: single quotes only please

File: web/modules/custom/user_role/user_role.module [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+use Drupal\Core\Form\FormStateInterface;
+/**
```
Comment (vivekpan27) [inline web/modules/custom/user_role/user_role.module:4]: There should be one empty line above this.

File: web/modules/custom/user_role/user_role.module [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * Implements hook_form_alter().
+ */
+function user_role_form_alter(&$form, FormStateInterface $form_state, $form_id) {
```
Comment (vivekpan27) [inline web/modules/custom/user_role/user_role.module:7]: `array &$form`.

Comment (Eshakundu01) [reply web/modules/custom/user_role/user_role.module:7]: In Drupal core, the hook is defined in this format. Then do I still need to define the datatype as you have mentioned?

**Pull request merged**